scripts/utilities/retry.js

```diff
@@ -0,0 +1,22 @@
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+function retry(functionToTry, maxAttempts = 3, delay = 1000) {
+  return new Promise(async (resolve, reject) => {
+    try {
+      functionToTry();
+      resolve();
+    } catch (error) {
+      const remainingAttempts = maxAttempts - 1;
+      if (remainingAttempts > 0) {
+        await sleep(delay);
+        await retry(functionToTry, remainingAttempts, delay);
+      }
+
+      reject(error);
```

Comment:
This catch block worries me a little bit.

The `reject` gets executed regardless of the output of the retry, and since it returns a new Promise I don't think a resolve output actually exits out of the entire call stack as you would expect.

Can you test that when the first request is errors out and is retried, but the second one succeeds, the original `retry` call still terminates with the `resolve` block instead of the `catch` block?

Reply:
Thanks @amrocha. You are very right! I wrote a quick test:

```js
let i = 0;
const c = () => {
  console.log(i);
  if (i < 1) {
    i += 1;
    throw new Error('c');
  } else {
    console.log('c');
  }
};

retry(c);
```

The output of this is:
```shell
0
1
c
(node:33745) UnhandledPromiseRejectionWarning: Error: c
```

I updated the code to fix up the output to resolve if it passed the retry function.
```shell
0
1
c
```